src/Storages/ExternalDataSourceConfiguration.cpp

```diff
@@ -93,6 +86,53 @@ void ExternalDataSourceConfiguration::set(const ExternalDataSourceConfiguration
     addresses_expr = conf.addresses_expr;
 }
 
+namespace
+{
+void initExternalDataSourceConfiguration(ExternalDataSourceConfiguration & configuration)
```

Comment:
`ExternalDataSourceConfiguration` (as well as all code in `ExternalDataSourceConfiguration.cpp/.h`) will soon be deleted completely in favour of code in `Common/NamedCollections/`, it is not deleted yet because dictionaries still use it, for storages I already replaced it to the new code. So it would be better if you could apply the implementation from `Common/NamedCollections/` instead (If you do this, ClickHouse dictionary source will automatically support named collections created from sql (https://github.com/ClickHouse/ClickHouse/pull/43252), better grants (https://github.com/ClickHouse/ClickHouse/pull/46241)). See `Common/NamedCollections/NamedCollections.h` and how it is used for storages.

Reply:
Hello @kssenii , I think I can switch Dictionaries to Common/NamedCollections by next Wednesday (Feb 22).
Not 100% sure about getExternalDataSourceConfigurationByPriority used for PostgreSQL though.
Let me know if it is Ok.
If you plan to work on this, I'd better wait while you are done to avoid collisions.

Reply:
Ok, haven't started, I'll do today.

Reply:
Hello @kssenii , let me know if I can be of help with this issue.

Reply:
Sorry, was (and now, till tomorrow to be more precise) on vacation, so did not finish yet. Will do the mentioned PR right after this one https://github.com/ClickHouse/ClickHouse/pull/46647 is merged, as it blocks changes for dictionaries.

Reply:
@kssenii , looks like blocking PR is merged. Could you resume this one please?